spec/requests/api/sessions/create_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+
+RSpec.describe 'POST api/v1/users/sign_in', type: :request do
+  let(:user) { create(:user) }
+  let(:params) do
+    {
+      user: {
+        email: user.email,
+        password: user.password
+      }
+    }
+  end
+
+  subject(:sign_in) { post user_session_path, params: params, as: :json }
+
+  context 'using valid sign in data' do
+    it 'returns success' do
+      sign_in
+      expect(response).to be_successful
+    end
+
+    it 'returns user data' do
+      sign_in
+      expect(json).to include_json(
+        user: {
+          id: user.id,
+          first_name: user.first_name,
+          last_name: user.last_name,
+          gender: user.gender,
+          email: user.email
+        }
+      )
+    end
+
+    it 'returns valid token authentication data' do
+      sign_in
+      token = response.header['access-token']
+      client = response.header['client']
+      expect(user.reload.valid_token?(token, client)).to be_truthy
+    end
+
+    it 'adds user token' do
+      headers = auth_headers
+      expect { sign_in }.to change { user.reload.tokens.size }.by(1)
+    end
+  end
+
+  context 'using invalid sign in data' do
+    before do
+      user.password = 'invalidPassword'
```

Comment:
question: how this works? does this update the user in the db? 🤔

Reply:
The value 'invalidPassword' is never saved. The value is set after the creation of the user, so the wrong password is sent when trying to log in. I'll make some changes to make it more readable.

Reply:
As we talked today, a better approach is:
in the main describe:
```
  let!(:user) { create(:user) }
  let(:user_emal) { user.email }
  let(:user_password) { user.password }
  let(:params) do
    {
      user: {
        email: user_email,
        password: user_password
      }
    }
  end
```
and in the context ```using invalid sign in data```:
```
let(:user_password) { 'invalidPassword' }
```